testlapack/dgebd2.go

```diff
@@ -0,0 +1,69 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"math"
+	"math/rand"
+	"testing"
+)
+
+type Dgebd2er interface {
+	Dgebd2(m, n int, a []float64, lda int, d, e, tauq, taup, work []float64)
+}
+
+func Dgebd2Test(t *testing.T, impl Dgebd2er) {
+	for _, test := range []struct {
+		m, n, lda int
+	}{
+		{3, 4, 0},
+		{4, 3, 0},
+		{3, 4, 10},
+		{4, 3, 10},
+	} {
+		m := test.m
+		n := test.n
+		lda := test.lda
+		if lda == 0 {
+			lda = n
+		}
+		nb := min(m, n) // name parallel with Dlabrd.
+		a := make([]float64, m*lda)
+		for i := range a {
+			a[i] = rand.NormFloat64()
+		}
+		d := make([]float64, nb)
```

Comment:
Maybe these all together warrant a helper to return poisoned []float64 of a given length.